pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -286,10 +287,15 @@ static void runHapiJdbcFetch(
     // Get the resource count for each resource type and distribute the query workload based on
     // batch size.
     HashMap<String, Integer> resourceCount =
-        (HashMap<String, Integer>) fhirSearchUtil.searchResourceCounts(options.getResourceList());
+        (HashMap<String, Integer>)
+            fhirSearchUtil.searchResourceCounts(options.getResourceList(), options.getSince());
 
     for (String resourceType : options.getResourceList().split(",")) {
       int numResources = resourceCount.get(resourceType);
+      if (numResources == 0) {
+        continue;
```

Comment:
how about we make the if statement set the `foundResource` variable to `true` if the number of resources returned is greater than 0

Reply:
Hmm, the point of the if statement is not just about `foundResource`; its main purpose is to avoid making a pipeline for any resource type where there are zero instances to fetch.

Reply:
Oh yes! I didn't notice the for loop contains the pipeline creation as well